commit 5d6b7b805ebc544ce95cd6c8238285e1ea3e820e (refs/changes/83/1983/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-05-22T14:28:19+02:00 (2 years, 5 months ago)
    
    TOOLS-1784 sdcadm up moray in a setup with non-HA moray fails to bring moray up

diff --git a/lib/procedures/index.js b/lib/procedures/index.js
index b40e7d8..0b3599f 100644
--- a/lib/procedures/index.js
+++ b/lib/procedures/index.js
@@ -26,17 +26,22 @@ var errors = require('../errors'),
     UsageError = errors.UsageError;
 
 // --- HA ready services
-var HA_READY_SVCS = [
+
+// Here "SIMPLE" means those that can be upgraded with the
+// UpdateStatelessServices.
+var HA_READY_SIMPLE_SVCS = [
     'adminui',
     'cloudapi',
     'cmon',
     'mahi',
-    'moray',
     'nat',
     'papi',
     'portolan',
     'workflow'
 ];
+var ALL_HA_READY_SVCS = HA_READY_SIMPLE_SVCS.concat([
+    'moray'
+]);
 
 // --- Disallowed agents
 var DISALLOWED_AGENTS = ['provisioner', 'heartbeater', 'zonetracker'];
@@ -172,10 +177,10 @@ function coordinatePlan(opts, cb) {
          */
         function updateSimpleServices(_, next) {
             var simpleServices = [
-                'adminui', 'amon', 'amonredis', 'assets', 'ca',
+                'amon', 'amonredis', 'assets', 'ca',
                 'cnapi', 'cns', 'dhcpd', 'docker', 'fwapi', 'manta',
                 'napi', 'rabbitmq', 'redis', 'sdc', 'vmapi'
-            ].concat(HA_READY_SVCS);
+            ].concat(HA_READY_SIMPLE_SVCS);
             var handle = [];
             var remaining = [];
             var errs = [];
@@ -217,7 +222,8 @@ function coordinatePlan(opts, cb) {
                             change.insts = svcInsts;
                             handle.push(change);
                         } else {
-                            if (~HA_READY_SVCS.indexOf(change.service.name)) {
+                            if (~HA_READY_SIMPLE_SVCS.indexOf(
+                                change.service.name)) {
                                 var chInsts = forceSameImage ? svcInsts :
                                         svcInsts.filter(function (ins) {
                                             return (ins.image !==
@@ -628,7 +634,8 @@ function coordinatePlan(opts, cb) {
 
                     var sName = change.service.name;
                     var allowed = ((avoid.indexOf(sName) === -1) &&
-                        (HA_READY_SVCS.indexOf(sName) !== -1 || change.force));
+                        (ALL_HA_READY_SVCS.indexOf(sName) !== -1 ||
+                            change.force));
 
                     if (allowed) {
                         log.debug({
